From fa70ee1cd78e2b161545bc47a1c1083063030f77 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Mon, 11 Oct 2021 10:52:43 +0200
Subject: [PATCH 1/5] renderer/native: Log render mode per device

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/2051>
(cherry picked from commit e8249a572d669c4c0a8464d6bce556b2cbaca4ef)
(cherry picked from commit fe0ea79b83256d80f1dee1e4a49c94a5d6fd18a6)
---
 src/backends/native/meta-renderer-native.c | 30 ++++++++++++++++++++++
 1 file changed, 30 insertions(+)

diff --git a/src/backends/native/meta-renderer-native.c b/src/backends/native/meta-renderer-native.c
index 735c37202..f92f648e5 100644
--- a/src/backends/native/meta-renderer-native.c
+++ b/src/backends/native/meta-renderer-native.c
@@ -1808,6 +1808,24 @@ meta_renderer_native_create_renderer_gpu_data (MetaRendererNative  *renderer_nat
   return NULL;
 }
 
+static const char *
+renderer_data_mode_to_string (MetaRendererNativeMode mode)
+{
+  switch (mode)
+    {
+    case META_RENDERER_NATIVE_MODE_GBM:
+      return "gbm";
+    case META_RENDERER_NATIVE_MODE_SURFACELESS:
+      return "surfaceless";
+#ifdef HAVE_EGL_DEVICE
+    case META_RENDERER_NATIVE_MODE_EGL_DEVICE:
+      return "egldevice";
+#endif
+    }
+
+  g_assert_not_reached ();
+}
+
 static gboolean
 create_renderer_gpu_data (MetaRendererNative  *renderer_native,
                           MetaGpuKms          *gpu_kms,
@@ -1822,6 +1840,18 @@ create_renderer_gpu_data (MetaRendererNative  *renderer_native,
   if (!renderer_gpu_data)
     return FALSE;
 
+  if (gpu_kms)
+    {
+      g_message ("Created %s renderer for '%s'",
+                 renderer_data_mode_to_string (renderer_gpu_data->mode),
+                 meta_gpu_kms_get_file_path (gpu_kms));
+    }
+  else
+    {
+      g_message ("Created %s renderer without GPU",
+                 renderer_data_mode_to_string (renderer_gpu_data->mode));
+    }
+
   g_hash_table_insert (renderer_native->gpu_datas,
                        gpu_kms,
                        renderer_gpu_data);
-- 
2.35.1

